src/webserver.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Provide the GunicornWebserver class to represent the gunicorn server."""
+import logging
+import pathlib
+import signal
+import typing
+
+from ops.model import Container
+from ops.pebble import ExecError, PathError
+
+from charm_state import CharmState
+from charm_types import ExecResult
+from consts import FLASK_SERVICE_NAME
+from exceptions import CharmConfigInvalidError
+
+logger = logging.getLogger(__name__)
+
+
+class GunicornWebserver:
+    """A class representing a Gunicorn web server.
+
+    Attrs:
+        command: the command to start the Gunicorn web server.
+
+    """
+
+    def __init__(self, charm_state: CharmState, flask_container: Container):
+        """Initialize a new instance of the GunicornWebserver class.
+
+        Args:
+            charm_state: The state of the charm that the GunicornWebserver instance belongs to.
+            flask_container: The Flask container in this charm unit.
+        """
+        self._charm_state = charm_state
+        self._flask_container = flask_container
+
+    @property
+    def _config(self) -> str:
```

Comment:
I'm wondering if jinja templating could be used here :)

Reply:
This problem has been resolved in the last PR, thanks!